azkaban-common/src/main/java/azkaban/project/ProjectFileHandler.java

```diff
@@ -46,7 +47,8 @@ public ProjectFileHandler(
       final int numChunks,
       final byte[] MD5Hash,
       final Set<Dependency> startupDependencies,
-      final String resourceId) {
+      final String resourceId,
```

Comment:
this seems to be changing public constructor. recommended approach for such situations is to create a new constructor with expanded set of vars and wrap it. eg: 

Change THIS: 
public foo(x, y) {
 // yada yada logic
}

TO: 

public foo(x, y) {
 foo(x, y, null);
}
public foo(x, y, z) {
 // yada yada logic
}

Reply:
Agreed. However, that approach is good when the constructor is called at many places. In this instance, it is called exactly at one place and the change is required there only.